### PR TITLE
Ignore deleted build configuration

### PR DESCRIPTION
**Pull request by blindpirate**
If a build configuration is deleted or renamed, the app will crash on
the 404 error. Now we capture the exception and ignore it.

### CODE REVIEW

**Comment by blindpirate** [comment]
Like this:

```
org.jetbrains.teamcity.rest.TeamCityConversationException: Failed to connect to https://builds.gradle.org/guestAuth/app/rest/buildTypes/id:Gradle_Release_Check_Stage_ReadyforMerge_Trigger: 404  Responding with error, status code: 404 (Not Found).
Details: jetbrains.buildServer.server.rest.errors.NotFoundException: No build type nor template is found by id 'Gradle_Release_Check_Stage_ReadyforMerge_Trigger'.
Could not find the entity requested. Check the reference is correct and the user has permissions to access the entity.
	at org.jetbrains.teamcity.rest.TeamCityInstanceImpl$service$3.handleError(implementation.kt:156)
	at org.jetbrains.teamcity.rest.TeamCityInstanceImpl$service$3.handleError(implementation.kt:105)
	at retrofit.RestAdapter$RestHandler.invoke(RestAdapter.java:242)
	at com.sun.proxy.$Proxy0.buildConfiguration(Unknown Source)
	at org.jetbrains.teamcity.rest.BuildConfigurationImpl.fetchFullBean(implementation.kt:843)
	at org.jetbrains.teamcity.rest.BuildConfigurationImpl.fetchFullBean(implementation.kt:839)
	at org.jetbrains.teamcity.rest.BaseImpl$fullBean$2.invoke(implementation.kt:650)
	at org.jetbrains.teamcity.rest.BaseImpl$fullBean$2.invoke(implementation.kt:629)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at org.jetbrains.teamcity.rest.BaseImpl.getFullBean(implementation.kt)
	at org.jetbrains.teamcity.rest.BuildConfigurationImpl.getName(implementation.kt:1791)
	at org.gradle.devprod.collector.teamcity.TeamcityClientService.verifyBuildConfigurationId(TeamcityClientService.kt:68)
	at org.gradle.devprod.collector.teamcity.TeamcityClientServiceKt.main(TeamcityClientService.kt:23)
	at org.gradle.devprod.collector.teamcity.TeamcityClientServiceKt.main(TeamcityClientService.kt)
Caused by: retrofit.RetrofitError: 404 
	at retrofit.RetrofitError.httpError(RetrofitError.java:40)
	at retrofit.RestAdapter$RestHandler.invokeRequest(RestAdapter.java:388)
	at retrofit.RestAdapter$RestHandler.invoke(RestAdapter.java:240)
	... 11 more
```

File: teamcity-export/src/main/kotlin/org/gradle/devprod/collector/teamcity/TeamcityClientService.kt [hunk 1]
```diff
@@ -42,18 +42,35 @@ class TeamcityClientService(
...
+                    .fromConfiguration(buildConfiguration)
+                    .includeCanceled()
+                    .includeFailed()
+                    .withAllBranches()
+                    .since(Instant.now().minus(5, ChronoUnit.DAYS))
+                    .all()
+                    .mapNotNull { it.toTeamCityBuild() }
+            }
+
+    private fun verifyBuildConfigurationId(buildConfigurationId: String): BuildConfigurationId? {
```
Comment (wolfs) [inline teamcity-export/src/main/kotlin/org/gradle/devprod/collector/teamcity/TeamcityClientService.kt:60]: I am not sure about this change. Why do we have a build configuration id which can't be found on TC in the first place? It seems like the job should fail in this case.

What I'd like to have is that the health check marks the app as unhealthy if one of these jobs fails.

Comment (blindpirate) [reply teamcity-export/src/main/kotlin/org/gradle/devprod/collector/teamcity/TeamcityClientService.kt:60]: This is aftermath of this change: https://github.com/gradle/gradle/commit/f0d9291c04b90b59445041eaa75b2ee744162586

I created a new PR to remove it: https://github.com/gradle/bt-dev-prod-data-collector/pull/23

However, as a general problem, sometimes we may change build configuration on `master` and `release` independently, which might cause the difference. I wonder if we need this in the future.

**Comment by blindpirate** [comment]
Superseded by https://github.com/gradle/bt-dev-prod-data-collector/pull/23

**Pull request closed without merging**